be/src/exprs/vectorized/encryption_functions.cpp

```diff
@@ -199,4 +200,141 @@ ColumnPtr EncryptionFunctions::md5(FunctionContext* ctx, const Columns& columns)
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
+Status EncryptionFunctions::sha2_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    if (scope != FunctionContext::FRAGMENT_LOCAL) {
+        return Status::OK();
+    }
+
+    if (!context->is_constant_column(1)) {
+        return Status::InternalError("sha2 just support const value");
+    }
+
+    ColumnPtr column = context->get_constant_column(1);
+    auto hash_length = ColumnHelper::get_const_value<TYPE_INT>(column);
+
+    ScalarFunction function;
+    if (hash_length == 224) {
```

Comment:
According to https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_sha2

0 is also valid length which equivalent to 256.

And when the hash_length is not valid, MySQL will return null for that record.